fda/functional_data.py

```diff
@@ -125,9 +117,185 @@ def domain_range(self):
         """
         pass
 
+    def _reshape_eval_points(self, eval_points, evaluation_aligned):
+        """Convert and reshape the eval_points to ndarray with the corresponding
+        shape.
+
+        Args:
+            eval_points (array_like): Evaluation points to be reshaped.
+            evaluation_aligned (bool): Boolean flag. True if all the samples
+                will be evaluated at the same evaluation_points.
+
+        Returns:
+            (np.ndarray): Numpy array with the eval_points, if
+            evaluation_aligned is True with shape `number of evaluation points`
+            x `ndim_domain`. If the points are not aligned the shape of the
+            points will be `nsamples` x `number of evaluation points`
+            x `ndim_domain`.
+        """
+
+        # Case evaluation of a scalar value, i.e., f(0)
+        if numpy.isscalar(eval_points):
+            eval_points = [eval_points]
+
+        # Creates a copy of the eval points, and convert to np.array
+        eval_points = numpy.array(eval_points)
+
+        if evaluation_aligned: # Samples evaluated at same eval points
+
+            eval_points = eval_points.reshape((eval_points.shape[0],
```

Comment:
One documentation example is failing at this line

Reply:
Do you have the example that is failing? In my computer and in github all the test are ok.

Reply:
It is one of the documentation examples:

```
WARNING: /home/docs/checkouts/readthedocs.org/user_builds/fda/checkouts/feature-fdata/examples/plot_shift_registration_basis.py failed to execute correctly: Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/fda/checkouts/feature-fdata/examples/plot_shift_registration_basis.py", line 50, in <module>
    fd_registered = fda.registration.shift_registration(fd_basis)
  File "/home/docs/checkouts/readthedocs.org/user_builds/fda/envs/feature-fdata/lib/python3.6/site-packages/fda-0.1-py3.6.egg/fda/registration.py", line 396, in shift_registration
    **kwargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/fda/envs/feature-fdata/lib/python3.6/site-packages/fda-0.1-py3.6.egg/fda/registration.py", line 307, in shift_registration_deltas
    extrapolation=extrapolation, keepdims=False)
  File "/home/docs/checkouts/readthedocs.org/user_builds/fda/envs/feature-fdata/lib/python3.6/site-packages/fda-0.1-py3.6.egg/fda/functional_data.py", line 340, in evaluate
    eval_points = self._reshape_eval_points(eval_points, aligned_evaluation)
  File "/home/docs/checkouts/readthedocs.org/user_builds/fda/envs/feature-fdata/lib/python3.6/site-packages/fda-0.1-py3.6.egg/fda/functional_data.py", line 147, in _reshape_eval_points
    self.ndim_domain))
ValueError: cannot reshape array of size 3015 into shape (15,1)
```

Reply:
Opps, I had the documentation in cache. I forgot to change the call to evaluate. Fixed in 49e2092.

 In the next PR I will add test for extrapolation and registration to avoid problems like this (an write/review the corresponding documentation).